api_payments.yaml

```diff
@@ -0,0 +1,286 @@
+swagger: "2.0"
+info:
+  version: 1.0.0
+  title: Proxy PagoPA
+  description: Proxy for PagoPA payment services
+host: localhost
+basePath: /api/v1
+schemes:
+  - https
+paths:
+  "/payment/check":
+    x-swagger-router-controller: PaymentController
+    get:
+      operationId: checkPaymentToPagoPa
+      summary: Check Payment
+      description: Retrieve information about a payment (qrcode)
+      parameters:
+        - in: body
+          name: body
+          schema:
+            $ref: "#/definitions/PaymentsCheckRequest"
+          x-examples:
+            application/json:
+              codiceIdRPT:
+                CF: "TMMEXQ60A10Y526X"
+                AuxDigit: 1
+                CodIUV: 123456789012345
+                CodStazPA: 56
+      responses:
+        '200':
+          description: Payment information retrieved
+          schema:
+            "$ref": "#/definitions/PaymentsCheckResponse"
+          examples:
+            application/json:
+              importoSingoloVersamento: 200,
+              codiceContestoPagamento: "ABC123"
+        "400":
+          description: Input provided or pagoPa response are invalid
+        "500":
+          description: PagoPa services are not available or request is rejected
+  "/payment/activation":
+    x-swagger-router-controller: PaymentController
+    get:
+      operationId: activatePaymentToPagoPa
+      summary: Activate Payment
+      description: Require a lock (activation) for a payment (qrcode)
+      parameters:
+        - in: body
+          name: body
+          schema:
+            $ref: "#/definitions/PaymentsActivationRequest"
+          x-examples:
+            application/json:
+              codiceIdRPT:
+                CF: "TMMEXQ60A10Y526X"
+                AuxDigit: 1
+                CodIUV: 123456789012345
+                CodStazPA: 56
+              importoSingoloVersamento: 200
+              codiceContestoPagamento: "ABC123"
+      responses:
+        '200':
+          description: Payment Activation process started
+          schema:
+            "$ref": "#/definitions/PaymentsActivationResponse"
+          examples:
+            application/json:
+              importoSingoloVersamento: 200
+        "400":
+          description: Input provided or pagoPa response are invalid
+        "500":
+          description: PagoPa services are not available or request is rejected
+  "/payment/activation/check/{codiceContestoPagamento}":
+    x-swagger-router-controller: PaymentController
+    get:
+      operationId: checkPaymentActivationStatusFromDB
+      summary: Check Activation status
+      description: Check the activation status to retrieve the paymentId
+      parameters:
+        - name: codiceContestoPagamento
+          in: path
+          type: string
+          required: true
+          description: The session token for PagoPa
```

Comment:
is this the session token ?

Reply:
Maybe better to call it Session Id (as I did in other parts)

Reply:
what is this session ID? PagoPA APIs don't have a concept of session (?)

Reply:
When App do his first request (verificaRPT), a codiceContestoPagamento is generated.
Any next request should be done using the same codiceContestoPagamento.

It's necessary to retrieve the async response for attivaRPT (a paymentId is related to a codiceContestoPagamento).
I think it's like a transaction\session id. Not?

Reply:
ok I suggest to keep strict pairing with the pagoPA nomenclature or else confusion may arise - you are right this is more like a transaction ID - the use of `session` led me to think that there was an authentication step for calling the API